src/daggy.js

```diff
@@ -131,6 +131,26 @@ const arrToString = (arr) => {
   return str + ')'
 }
 
+const makeConstructor= (fields, proto) => {
+  switch (fields.length) {
+    case 1: return function(a) { return makeValue(fields, proto, arguments) }
```

Comment:
yes that's intentional this this changes we are not making constructors curried. what we are doing is making sure that length property of the constructor functions reflect number of arguments they expect so that one could just use R.curry for example and things will work.